modules/video/test/test_ecc.cpp

```diff
@@ -96,6 +96,44 @@ double CV_ECC_BaseTest::computeRMS(const Mat& mat1, const Mat& mat2){
 }
 
 
+class CV_ECC_Test_Compute : public cvtest::BaseTest
+{
+public:
+    CV_ECC_Test_Compute();
+protected:
+    void run(int);
+    bool testCompute(int);
+};
+
+CV_ECC_Test_Compute::CV_ECC_Test_Compute(){
+}
+
+bool CV_ECC_Test_Compute::testCompute(int)
+{
```

Comment:
Lets avoid using of legacy testing framework.

Just use Google Test:
```
TEST(Video_ECC_Test_Compute, accuracy)
{
    Mat testImg = (Mat_<float>(3, 3) << 1, 0, 0, 1, 0, 0, 1, 0, 0);
    Mat warpedImage = (Mat_<float>(3, 3) << 0, 1, 0, 0, 1, 0, 0, 1, 0);
    Mat_<unsigned char> mask = Mat_<unsigned char>::ones(testImg.rows, testImg.cols);
    double ecc = computeECC(warpedImage, testImg, mask);

    EXPECT_NEAR(ecc, -0.5f, 1e-5f);
}
```

Reply:
Done: Thanks for writing out the Google Test version.